geth/api/backend.go

```diff
@@ -29,7 +29,7 @@ type StatusBackend struct {
 	accountManager common.AccountManager
 	txQueueManager common.TxQueueManager
 	jailManager    common.JailManager
-	notification   common.Notifier
+	notifier       common.NotifierConstructor
```

Comment:
Why changing it here?

Reply:
I've considered 'no state == no race' and changed constructor to higher order function to make new Notify object every time. Notify object is small and we can make it fast and safe, with no races risk.